peer_test.go

```diff
@@ -588,3 +594,152 @@ func TestPeerChannelClosureFeeNegotiationsInitiator(t *testing.T) {
+		activeNetParams.Params,
+	)
+	if err != nil {
+		t.Fatalf("invalid delivery address: %v", err)
+	}
+
+	script, err := txscript.PayToAddrScript(deliveryAddr)
+	if err != nil {
+		t.Fatalf("cannot create script: %v", err)
+	}
+
+	// setShutdown is a function which sets the upfront shutdown address for
+	// the local channel.
+	setShutdown := func(a, b *channeldb.OpenChannel) {
+		a.LocalShutdownScript = script
+		b.RemoteShutdownScript = script
+	}
+
+	tests := []struct {
+		name string
+
+		// update is a function used to set values on the channel set up for the
+		// test. It is used to set values for upfront shutdown addresses.
+		update func(a, b *channeldb.OpenChannel)
+
+		// userCloseAddress is the address specified by the user.
+		userCloseAddress lnwire.DeliveryAddress
+
+		// expectedAddress is the address we expect to be set on the shutdown
+		// message.
+		expectedAddress lnwire.DeliveryAddress
+
+		// expectErr is true is we expect the request to close the channel to
+		// fail.
+		expectErr bool
+	}{
+		{
+			name:             "User set address",
+			update:           noUpdate,
+			userCloseAddress: script,
+			expectedAddress:  script,
+		},
+		{
+			name:   "No user set address",
+			update: noUpdate,
+		},
+		{
+			name:            "Shutdown set, no user address",
+			update:          setShutdown,
+			expectedAddress: script,
+		},
+		{
+			name:             "Shutdown set, user address matches",
+			update:           setShutdown,
+			userCloseAddress: script,
+			expectedAddress:  script,
+		},
+		{
+			name:             "Shutdown set, user address different",
+			update:           setShutdown,
+			userCloseAddress: []byte("different addr"),
+			expectErr:        true,
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			notifier := &mockNotfier{
+				confChannel: make(chan *chainntnfs.TxConfirmation),
+			}
+			broadcastTxChan := make(chan *wire.MsgTx)
+
+			// Open a channel.
+			initiator, initiatorChan, _, cleanUp, err := createTestPeer(
+				notifier, broadcastTxChan, test.update,
+			)
+			if err != nil {
+				t.Fatalf("unable to create test channels: %v", err)
+			}
+			defer cleanUp()
+
+			// Request initiator to cooperatively close the channel, with
+			// a specified delivery address.
+			updateChan := make(chan interface{}, 1)
+			errChan := make(chan error, 1)
+			chanPoint := initiatorChan.ChannelPoint()
+			closeCommand := htlcswitch.ChanClose{
+				CloseType:      htlcswitch.CloseRegular,
+				ChanPoint:      chanPoint,
+				Updates:        updateChan,
+				TargetFeePerKw: 12500,
+				DeliveryAddr:   test.userCloseAddress,
+				Err:            errChan,
+			}
+
+			// Send the close command for the correct channel and check that a
+			// shutdown message is sent.
+			initiator.localCloseChanReqs <- &closeCommand
+
+			var msg lnwire.Message
+			select {
+			case outMsg := <-initiator.outgoingQueue:
+				msg = outMsg.msg
+			case <-time.After(time.Second * 5):
```

Comment:
Why 5 sec? Should this be a timeout const?

Reply:
tbh, this PR was originally a contributor PR which I took over, but I think they just followed the existing pattern in this file. I'll make a timeout var though, nice idea 👍